crates/handlers/src/admin/v1/users/deactivate.rs

```diff
@@ -76,15 +96,17 @@ pub async fn handler(
     }: CallContext,
     NoApi(mut rng): NoApi<BoxRng>,
     id: UlidPathParam,
+    body: Option<Json<Request>>,
 ) -> Result<Json<SingleResponse<User>>, RouteError> {
+    let Json(params) = body.unwrap_or_default();
     let id = *id;
     let mut user = repo
         .user()
         .lookup(id)
         .await?
         .ok_or(RouteError::NotFound(id))?;
 
-    if user.locked_at.is_none() {
+    if !params.skip_lock && user.locked_at.is_none() {
```

Comment:
I'm not sure what the better behaviour is. Obviously, deactivation is expansive, so you kinda want it to happen in the background. At the same time, if we don't flip the deactivated bit on the user right away, it will still be valid for some time?

Maybe this should just flip the deactivated bit right away and schedule a deactivation job?